.github/workflows/CI-system-test.yml

```diff
@@ -0,0 +1,65 @@
+name: System Tests and Publish Docker Image
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - feat/web3-943-post-merge-pipeline
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+-[0-9]+.[0-9]+.[0-9]+*"
```

Comment:
There is already a `CI-tag-orchestrator.yml` that is triggered w the same conditions
https://github.com/HorizenLabs/zkVerify/blob/42c361612e7bd12cd0fbd7d39f4f0e3d2be544de/.github/workflows/CI-tag-orchestrator.yml#L3-L6

It can be combined into one; otherwise the docker.sh will run twice for every tag.

Reply:
Other option would be to have 2 workflows: one for push to main, one for pushed tag.

For push to main, set env variables as done here (lines 58-61) and no need to run `setup_env.sh`
For pushed tag, add e2e test execution, and use `setup_env.sh` for env var setup